docs: en 08 translation (#968)
ipfs-force-community/damocles: docs/en/08.support-for-snapdeal.md

# Support for SnapDeal

## Brief Overview of SnapDeal
`SnapDeal` is a sector upgrade proposal introduced in [FIP-19](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0019.md) and launched on `network15`. Compared to previous upgrade proposals that required a complete re-encapsulation process, SnapDeal is relatively lightweight. Its cost is approximately:
- Completing an `add piece` task
- Completing a `tree d` task
- Completing a `snap_encode` task, which costs about the same as a `pc2` task
- Completing a `snap_prove` task, which costs about the same as a `c1` + `c2` task

Therefore, SnapDeal is attractive for both new real data storage requirements and the conversion of existing `CC sectors`.

> ## damocles Support for SnapDeal

Author: 也许我的文件名称也得修改了

Throughout this process, users only need to focus on importing and monitoring the availability of local candidate sectors, as all other processes will be automated.

## Example
Let's use an example of a production cluster on the `butterfly` network to demonstrate how to configure the SnapUp production scheme.

### Importing Candidate Sectors
Using the newly added `util sealer snap fetch` tool, you can import `CC sectors` that meet the condition (remaining lifetime greater than 180 days and satisfying the minimum lifetime requirement for deals) as local candidate sectors based on their deadlines.
```
./dist/bin/damocles-manager util sealer snap fetch 1153 3
2022-04-15T04:28:03.380Z        DEBUG   policy  policy/const.go:18      NETWORK SETUP   {"name": "butterfly"}
2022-04-15T04:28:03.401Z        INFO    cmd     internal/util_sealer_snap.go:53 candidate sectors fetched        {"available-in-deadline": 2, "added": 2}
```

### Monitoring Candidate Sector Availability
```
./dist/bin/damocles-manager util sealer snap candidates 1153
2022-04-15T04:28:13.955Z        DEBUG   policy  policy/const.go:18      NETWORK SETUP   {"name": "butterfly"}
deadline  count
3         2
```

> As seen, there are currently 2 `CC sectors` available in the `#3 deadline` as candidates for upgrade.

Reviewer: ```suggestion
In the above example, there are currently 2 `CC sectors` available in the `#3 deadline` as candidates for upgrade.
```

[[processors.snap_encode]]
envs = { FIL_PROOFS_USE_GPU_COLUMN_BUILDER = "1", FIL_PROOFS_USE_GPU_TREE_BUILDER = "1", CUDA_VISIBLE_DEVICES = "0", TMPDIR="/var/tmp/worker0" }

[[processors.snap_prove]]
envs = { CUDA_VISIBLE_DEVICES = "0" , TMPDIR="/var/tmp/worker0" }

[[processors.snap_encode]]
envs = { FIL_PROOFS_USE_GPU_COLUMN_BUILDER = "1", FIL_PROOFS_USE_GPU_TREE_BUILDER = "1", CUDA_VISIBLE_DEVICES = "0", TMPDIR="/var/tmp/worker1" }

[[processors.snap_prove]]
envs = { CUDA_VISIBLE_DEVICES = "1", TMPDIR="/var/tmp/worker1" }
```

The resource allocation for `snap_encode` can be similar to `pc2`, and for `snap_prove`, it can be similar to `c2`.

### Configuring damocles-manager
In `damocles-manager`, the required configuration mainly involves enabling `SnapUp` for specific `SPs`. Here is an example:

```
[[Miners]]
Actor = 1153
[Miners.Sector]
InitNumber = 0
MaxNumber = 10000
Enabled = true
EnableDeals = false

[Miners.SnapUp]
Enabled = true
Sender = "t1abjxfbp274xpdqcpuaykwkfb43omjotacm2p3za"
```

In this configuration:
- The settings in `[Miners.Sector]` do notSpecify the complete configuration details for the SnapUp production scheme. However, the key points to note are:

> - Set `Enabled = true` to enable SnapUp for the specified `SP`.

Reviewer: ```suggestion
```

- Specify the `Sender` address, which will be used to send SnapUp messages.

> ### Starting the Production

Reviewer: ```suggestion
### Tips
```

> Once the configurations are in place, you can start the production process by running `damocles-manager` and `damocles-worker` with the appropriate configurations. The `damocles-worker` will automatically convert the `sealing_thread`s into `SnapUp` production plans and start processing the candidate sectors.

Reviewer: ```suggestion
- Considering the computing resources required by `snap_encode` and `snap_prove`, if you enable regular sector sealing and `SnapUp` at the same time in the same `damocles-worker` instance, you may encounter race condition of hardware resource. You can refer to 07.damocles-worker external executor configuration example (En doc to be updated).
```

> Please note that the provided example is a simplified version, and the actual configuration and setup may vary based on your specific requirements and environment.

Reviewer: ```suggestion
- Considering the deployment of sector persistent storage, the `damocles-worker` used for `SnapUp` needs to be able to have both read and write access to all persistent storage spaces (persist stores), and ensure that their names are consistent with those defined in `damocles-manager`.
- Based on the above two points, we recommend using a separate box specifically for `SnapUp` production to avoid the configuration, operation and maintenance complexity caused by mixing regular sectors and `SnapUp` sealing.
```

> ## Conclusion

Reviewer: ```suggestion
## Continuous Improvement
```

> Support for SnapDeal in the `damocles` production system provides an automated and efficient way to upgrade existing `CC sectors` to the SnapDeal sector format. By following the outlined steps and configuring the necessary components, users can easily initiate the SnapUp production scheme and leverage the benefits of SnapDeal on the Filecoin network.

Reviewer: ```suggestion
The improvement and optimization of the `SnapUp` solution are still in progress. Currently we are mainly focusing on:

- Convert semi-automatic candidate sector import to automatic mode, or provide equivalent operation and maintenance tools
- More candidate sector import rules, such as import by storage configuration
- Aggregation of on-chain messages to reduce costs
- Other optimizations and tooling that can simplify operation & maintenance, reduce costs, and improve efficiency
```
